src/Compiler/Checking/MethodCalls.fs

```diff
@@ -1492,14 +1503,9 @@ let AdjustCallerArgForOptional tcVal tcFieldInit eCallerMemberName (infoReader:
             | NotOptional ->
                 //  T --> Nullable<T> widening at callsites
                 if isOptCallerArg then errorR(Error(FSComp.SR.tcFormalArgumentIsNotOptional(), m))
-                if isNullableTy g calledArgTy then 
```

Comment:
Again, quite a lot of logic removed here. Does AdjustCallerArgExpr really cover all this? Is it just a simplification or a fix?

Reply:
See https://github.com/dotnet/fsharp/pull/13673#discussion_r949553104